### PR TITLE
feat(openapi-tooling): add a watch mode for generate command

### PR DESCRIPTION
**Pull request by aramissennyeydd**
## Hey, I just made a Pull Request!

The current workflow for using `package schema openapi generate` is pretty cumbersome and has been a pretty annoying part of working with OpenAPI schemas. Hoping to make it a bit nicer with a watch mode that you can run in the background and will regenerate the client/server for you.

Next steps here are better caching so we reduce the number of times we have to run the full generator model.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by aramissennyeydd** [comment]
This one may have been opened through summer reviews, would very much appreciate a review here 🙇 @camilaibs

File: packages/repo-tools/src/commands/index.ts [hunk 1]
```diff
@@ -58,6 +58,8 @@ function registerPackageCommand(program: Command) {
     .description(
       'Additional properties that can be passed to @openapitools/openapi-generator-cli',
     )
+    .option('--watch')
```
Comment (camilaibs) [inline packages/repo-tools/src/commands/index.ts:61]: I know that there is no tests for these commands, but as I haven't used them, I would really appreciate a test for the changes.

**Review by camilaibs — COMMENTED** [review]
Hi, sorry for the delay in reviewing and thanks for the contributions 🙏🏻 
Do you mind adding instructions on how we can run it locally? I haven't used this CLI before, if you provide me with an example, I would be happy to verify it locally 🙂 .

File: packages/repo-tools/src/commands/index.ts (hunk 1 above)
Comment (aramissennyeydd) [reply packages/repo-tools/src/commands/index.ts:61]: Is there prior art for a `--watch` style test? Having a bit of trouble handling the parallel updates with jest.

**Comment by aramissennyeydd** [comment]
@camilaibs Ah, sorry 😅  - you should be able to run `yarn generate --watch` from `plugins/catalog-backend` (or `yarn workspace @backstage/plugin-catalog-backend generate --watch` and then make changes to the `src/schema/openapi.yaml` and check that the model files updated correctly.

**Review by Rugvip — COMMENTED** [review]
Nice! 😁 

Tried it out a bit, works like a charm 👍. I think error display could be improved a bit though, commented on that separately.

I also ran into a `src/generated/.openapi-generator-ignore` file  being created for me when I had an error. Do you know what that's about? 😁

**Review by camilaibs — APPROVED** [review]
Thanks again! Could you please look at Patrik's comments? Otherwise, it looks great to me 🙂

File: packages/repo-tools/src/lib/runner.ts [hunk 2]
```diff
@@ -67,3 +67,7 @@ export async function runner(
 
   return resultsList;
 }
+
+export async function block() {
+  return new Promise(() => {});
+}
```
[the remark this answers is not in the document]
Comment (aramissennyeydd) [reply packages/repo-tools/src/lib/runner.ts:73]: @Rugvip Did this for the unit tests so they didn't infinitely block, should I drop the test?

File: packages/repo-tools/src/commands/package/schema/openapi/generate/index.ts [hunk 3]
```diff
@@ -25,10 +29,62 @@ export async function command(opts: OptionValues) {
...
+          opts.clientPackage,
+          opts.clientAdditionalProperties,
+          abortSignal,
+        ),
+      );
+    }
+    if (opts.server) {
+      promises.push(generateServer(abortSignal));
+    }
+    await Promise.all(promises);
```
[the remark this answers is not in the document]
Comment (aramissennyeydd) [reply packages/repo-tools/src/commands/package/schema/openapi/generate/index.ts:47]: Gotcha, I'll take a look

[the remark this answers is not in the document]
Comment (aramissennyeydd) [reply packages/repo-tools/src/commands/package/schema/openapi/generate/index.ts:47]: @Rugvip agreed, the Java errors from the generator are really rough. I updated this to check the spec before trying to send it to the various commands which I think helps here. I tried to add some indentation so it's more clear which message belongs to which generator (server or client). Can you take another look?

**Comment by aramissennyeydd** [comment]
@Rugvip Depending on when your process fails, we may not clean up some of the files we create for the generator - `.openapi-generator-ignore` is one of those. Full list [here](https://github.com/aramissennyeydd/backstage/blob/f9e72ba0e3d8dce28bb334735580b87fce5091c2/packages/repo-tools/src/commands/package/schema/openapi/generate/client.ts#L95-L100)

File: packages/repo-tools/src/commands/package/schema/openapi/generate/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/repo-tools/src/commands/package/schema/openapi/generate/index.ts:56]: Awesome, that's a huge improvement! 👏

File: packages/repo-tools/src/lib/runner.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rugvip) [reply packages/repo-tools/src/lib/runner.ts:73]: Ah alright, could've used a comment perhaps but that's fine

**Review by Rugvip — APPROVED** [review]
🚢 🚢 🚢

**Pull request merged**